lms/djangoapps/courseware/views/views.py

```diff
@@ -602,7 +603,8 @@ def get(self, request, course_id, tab_type, **kwargs):
             course = get_course_with_access(request.user, 'load', course_key)
             try:
                 # Render the page
-                tab = CourseTabList.get_tab_by_type(course.tabs, tab_type)
+                course_tabs = course.tabs + _get_dynamic_tabs(course, request.user)
```

Comment:
@stvstnfrd Yes, course.tabs does not include any dynamic tabs since they are not persisted. This meant that dynamic tabs can't use this view to render themselves since this view only takes static tabs into account. 

Another for us is to not use this view and instead to provide a separate URL and view for lti tabs. This might actually be the preferred option, but either way this seemed like a bug.

Reply:
@xitij2000, Can't we use ``get_course_tab_list`` instead? ``_get_dynamic_tabs`` seems to be a private function by convention. Or should we just make it public? (removing underscore?)

Reply:
I did try that but that too has issues with returning the full proper list of tabs. I didn't want to go down a huge rabbithole, and this works as expected for this situation. 

As for the private function, I guess it is private at the module level but it's being imported within the same package so I figured it's OK. That said I think I should probably remove the underscore.